test_task_discovery fails in the pressence of other ewoks apps

***In GitLab by @woutdenolf on Apr 20, 2025, 12:13 GMT+2:***

```
src/ewokscore/tests/test_task_discovery.py:145: in assert_tasks
    expected_task = _find_task(expected, task["task_identifier"], task["task_type"])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

tasks = [{'category': 'ewokscore', 'description': 'Check whether a value is too small', 'input_model': None, 'n_required_posit...core', 'description': 'Add 1 to the first argument', 'input_model': None, 'n_required_positional_inputs': 0, ...}, ...]
identifier = 'ewoksfluo.tasks.example_data.tasks.MeshSingleScanSingleDetector', task_type = 'class'

    def _find_task(tasks, identifier, task_type):
        for task in tasks:
            if task["task_identifier"] == identifier and task["task_type"] == task_type:
                return task
    
>       raise ValueError(f"Task {identifier} and type {task_type} not found")
E       ValueError: Task ewoksfluo.tasks.example_data.tasks.MeshSingleScanSingleDetector and type class not found

src/ewokscore/tests/test_task_discovery.py:140: ValueError
```

**Assignees:** @woutdenolf

*Migrated from GitLab: https://gitlab.esrf.fr/workflow/ewoks/ewokscore/-/issues/69*

--- a/src/ewokscore/tests/test_task_discovery.py
+++ b/src/ewokscore/tests/test_task_discovery.py
@@ -142,6 +142,8 @@ def _find_task(tasks, identifier, task_type):
 
 def assert_tasks(tasks, expected):
     for task in tasks:
+        if task["category"] != "ewokscore":
+            continue
         expected_task = _find_task(expected, task["task_identifier"], task["task_type"])
         for key, value in task.items():
             assert expected_task[key] == value

